paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/MultiTableCommittable.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.flink.sink;
+
+import org.apache.paimon.catalog.Identifier;
+
+/**
+ * MultiTableCommittable produced by {@link PrepareCommitOperator}. This type of Committable will
+ * only be produced by multiplexed operators that handles data from multiple tables. Thus, the
+ * database, table, and commit user of each table is included in the committable.
+ */
+public class MultiTableCommittable extends Committable {
```

Comment:
Can we use just create a field `Committable` instead of extending?
Extending will make the caller think it can be used as a Committable.

Reply:
The extending class is necessary here becase in the base case `PrepareCommitOperator`, we have the following signature:

```java
    protected abstract List<Committable> prepareCommit(boolean doCompaction, long checkpointId)
            throws IOException;
```

Therefore, the new operator should also produce a subclass of committable and use another custom serializer.

Reply:
I think it is OK to bring generic type to `PrepareCommitOperator`.
`PrepareCommitOperator<IN, OUT>`

Reply:
Done refactoring to PrepareCommitOperator<IN, OUT>. Please take a look.